server/base/src/main/java/org/apache/accumulo/server/util/Admin.java

```diff
@@ -114,6 +127,93 @@ static class PingCommand {
     List<String> args = new ArrayList<>();
   }
 
+  @Parameters(commandNames = "check",
+      commandDescription = "Performs checks for problems in Accumulo.")
+  public static class CheckCommand {
+    @Parameter(names = "list",
+        description = "Lists the different checks that can be run, the description of each check, and the other check(s) each check depends on.")
+    boolean list;
+
+    @Parameter(names = "run",
+        description = "Runs the provided check(s) (explicit list or regex pattern specified following '-p'), beginning with their dependencies, or all checks if none are provided.")
+    boolean run;
+
+    @Parameter(names = {"--name_pattern", "-p"},
+        description = "Runs all checks that match the provided regex pattern.")
+    boolean useRegex;
```

Comment:
Would it be possible to make the following change?  Seems that currently when pattern is specified it looks for the pattern in the `checks` parameter.

```suggestion
    @Parameter(names = {"--name_pattern", "-p"},
        description = "Runs all checks that match the provided regex pattern.")
    String pattern;
```

Reply:
Addressed in 3bd6ce48b6831fb08d6c5cc894699ee65d03eff9